decidim-core/lib/decidim/view_hooks.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module Decidim
+  # This class acts as a registry for view hooks. It saves paths to partials found in
+  # engines so they can be included in other engines. Each engine can have its own
+  # instance of this class, so that view hooks are namespaced instead of global.
+  #
+  # This mechanism is useful to extend the views of a given engine from other engines.
+  # For example, the homepage of Decidim is found on `decidim-core`, but it can be
+  # extended by other engines to show important info there. For example, an engine might
+  # want to extend the homepage to show highlighted participatory processes.
+  #
+  # In order to show view hooks, you can use something like this in your views:
+  #
+  #     Decidim.view_hooks # => an instance of this class
+  #     <% Decidim.view_hooks.get(:my_hook).each do |hook| %>
```

Comment:
Update this?

Reply:
yes, I need to update the docs, I was more concerned about the API. I guess you like it since you didn't comment on it :trollface: